app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt

```diff
@@ -464,26 +470,37 @@ abstract class MultiSelectFragment(private val matomoCategory: String) : Fragmen
     }
 
     private fun observeMediator(
-        mediator: MediatorLiveData<Pair<Int, Int>>,
+        mediator: MediatorLiveData<MultiSelectMediatorState>,
         fileCount: Int,
         type: BulkOperationType,
         destinationFolder: File?,
         dialog: Dialog? = null,
     ) {
-        mediator.observe(viewLifecycleOwner) { (success, total) ->
+        mediator.observe(viewLifecycleOwner) { (success, total, error) ->
             if (total == fileCount) {
                 dialog?.dismiss()
-                handleIndividualActionsResult(success, type, destinationFolder)
+                handleIndividualActionsResult(success, error, type, destinationFolder)
             }
         }
     }
 
-    private fun handleIndividualActionsResult(success: Int, type: BulkOperationType, destinationFolder: File?) {
-        val title = if (success == 0) {
-            getString(R.string.anErrorHasOccurred)
-        } else {
-            resources.getQuantityString(type.successMessage, success, success, destinationFolder?.name + "/")
+    private fun handleIndividualActionsResult(
+        success: Int,
+        errorCode: String?,
+        type: BulkOperationType,
+        destinationFolder: File?
+    ) {
+        val title = when (errorCode) {
+            "limit_exceeded_error" -> getString(R.string.errorFilesLimitExceeded)
```

Comment:
Maybe here, it's better to only use a `if else if else` instead of a `when` because we're only using a when for `errorCode` for one case.